src/core/src/pass/manager.cpp

```diff
@@ -89,85 +77,117 @@ class stopwatch {
     bool m_active = false;
     std::chrono::nanoseconds m_last_time = std::chrono::high_resolution_clock::duration::zero();
 };
+
+class Profiler {
+public:
+    Profiler(bool visualize, bool profile_pass_enable)
+            : m_visualize(visualize),
+              m_profile_pass_enable(profile_pass_enable) {}
+
+    void start_timer(const std::string& name) {
+        if (m_profile_pass_enable) {
+            stopwatches[name] = stopwatch();
+            stopwatches[name].start();
+        }
+    }
+
+    void stop_timer(const std::string& name, const std::string& msg) {
+        if (m_profile_pass_enable) {
+            auto& stopwatch = stopwatches.at(name);
+            stopwatch.stop();
+            cout << msg << setw(7) << stopwatch.get_milliseconds() << "ms"
```

Comment:
Could you please keep the `(pass_applied ? " + " : "   ")` in the message? It is super useful for me.

Reply:
I haven't deleted it 
<img width="295" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/621c2f85-8677-4e92-a5b4-fd3f0eecebff">
I will double check the output formatting

Reply:
Yes, I missed it, sorry.